packages/Interface/src/DTK_C_API.h

```diff
@@ -0,0 +1,211 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+#ifndef DTK_C_API_H
+#define DTK_C_API_H
+
+#include <DataTransferKit_config.hpp>
+
+#include <DTK_Types.h>
+
+#ifndef __cplusplus
+#include <stdbool.h> // for bool
+#include <stddef.h>  // for size_t
+#else
+#include <cstddef> // for size_t
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*! Return current DTK version. */
+extern const char *DTK_version();
+
+/*! Get current repository hash.
+ *
+ * Returns an error string if it is not a GIT repository
+ */
+extern const char *DTK_git_commit_hash();
+
+typedef void *DTK_Handle;
+
+typedef enum { DTK_SERIAL, DTK_OPENMP, DTK_CUDA } DTK_EXECUTION_SPACE;
+
+/*! Create a DTK handle. */
+extern DTK_Handle DTK_create( DTK_EXECUTION_SPACE space );
+
+/*! Destroy a DTK handle */
+extern void DTK_destroy( DTK_Handle *handle );
+
+/*! Initialize DTK */
+extern void DTK_initialize( int *argc, char ***argv );
+
+/*! Check whether DTK is initialized */
+extern int DTK_is_initialized();
+
+/*! Finalize DTK */
+extern void DTK_finalize();
+
+typedef enum {
+    DTK_NODE_LIST_SIZE_FUNCTION,
+    DTK_NODE_LIST_DATA_FUNCTION,
+    DTK_BOUNDING_VOLUME_LIST_SIZE_FUNCTION,
+    DTK_BOUNDING_VOLUME_LIST_DATA_FUNCTION,
+    DTK_POLYHEDRON_LIST_SIZE_FUNCTION,
+    DTK_POLYHEDRON_LIST_DATA_FUNCTION,
+    DTK_CELL_LIST_SIZE_FUNCTION,
+    DTK_CELL_LIST_DATA_FUNCTION,
+    DTK_MIXED_TOPOLOGY_CELL_LIST_SIZE_FUNCTION,
+    DTK_MIXED_TOPOLOGY_CELL_LIST_DATA_FUNCTION,
+    DTK_BOUNDARY_SIZE_FUNCTION,
+    DTK_BOUNDARY_DATA_FUNCTION,
+    DTK_DOF_MAP_SIZE_FUNCTION,
+    DTK_DOF_MAP_DATA_FUNCTION,
+    DTK_MIXED_TOPOLOGY_DOF_MAP_SIZE_FUNCTION,
+    DTK_MIXED_TOPOLOGY_DOF_MAP_DATA_FUNCTION,
+    DTK_FIELD_SIZE_FUNCTION,
+    DTK_PULL_FIELD_DATA_FUNCTION,
+    DTK_PUSH_FIELD_DATA_FUNCTION,
+    DTK_EVALUATE_FIELD_FUNCTION
+} DTK_FUNCTION_TYPE;
+
+/*! Register a function with DTK. */
+extern void DTK_set_function( DTK_Handle handle, DTK_FUNCTION_TYPE type,
+                              void ( *f )(), void *user_data );
+
+/*! Get the size parameters for building a node list. */
+typedef void ( *DTK_node_list_size_function )( void *user_data,
+                                               unsigned *space_dim,
+                                               size_t *local_num_nodes,
+                                               bool *has_ghosts );
```

Comment:
I would prefer to call them the style typically used in C. I don't think camel case is common there.